### PR TITLE
feat(eap): separate value types in endpoint

### PR DESCRIPTION
**Pull request by volokluev**
1. Add a type argument to the tags endpoint
2. Rename according to EAP terminology
3. Add `trace_item_name` to request payload (in the current case it's `eap_spans`)
4. Add endpoint for attrivute values given specific key

### CODE REVIEW

**Review by Zylphrex — APPROVED** [review]

File: proto/sentry_protos/snuba/v1alpha/endpoint_tags_list.proto [hunk 1]
```diff
@@ -3,24 +3,31 @@ syntax = "proto3";
...
     string name = 1;
-    Type type = 2;
+    AttributeKey.Type type = 2;
   }
   repeated Tag tags = 1;
 }
+
+message AttributeValuesRequest {
+  RequestMeta meta = 1;
+  string name = 3;
```
Comment (Zylphrex) [inline proto/sentry_protos/snuba/v1alpha/endpoint_tags_list.proto:26]: Ahh actually this is missing an argument for a substring match. i.e. if an user starts typing a value, we should pass it through and only return values that contain that substring.

Comment (volokluev) [reply proto/sentry_protos/snuba/v1alpha/endpoint_tags_list.proto:26]: Done

File: proto/sentry_protos/snuba/v1alpha/endpoint_tags_list.proto [hunk 2]
```diff
@@ -3,24 +3,33 @@ syntax = "proto3";
...
+import "sentry_protos/snuba/v1alpha/trace_item_attribute.proto";
 
 //A request for "which tags are available for these projects between these dates" - used for things like autocompletion
-message TagsListRequest {
+message TraceItemAttributesRequest {
   RequestMeta meta = 1;
   uint32 limit = 2;
   uint32 offset = 3;
+  string trace_item_name = 4;
+  AttributeKey.Type type = 5;
```
[the remark this answers is not in the document]
Comment (volokluev) [reply proto/sentry_protos/snuba/v1alpha/endpoint_tags_list.proto:13]: No I'll probably reject them on the backend. Think I should just make another attrtibute type definition?

**Review by Zylphrex — APPROVED** [review]

File: proto/sentry_protos/snuba/v1alpha/endpoint_tags_list.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zylphrex) [reply proto/sentry_protos/snuba/v1alpha/endpoint_tags_list.proto:13]: If there's the possibility of these types being added at some point, perhaps keep them around and just return an empty response.

**Pull request merged**